calblueprint/girls-write-now-node
# [story] Implement job to push story to Supabase

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
In this PR, I have created the basic framework for porting stories information from the NPO's Wordpress database into our personal Supabase database, specifically pushing to the stories schema. In this PR, I have created basic getAll functions for stories, collections, genre mediums, topics, and coauthor api endpoints. However, we are mainly using the getAllStories function, and running this output into a filterStories() function that will filter out story outputs that are not text based. I have also written a basic insertStory function that connects to our Supabase database and pushes the filtered story objects based on the outlined schema. 

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## Relevant Links
N/A

### Notion Sprint Task
https://www.notion.so/calblueprint/story-Implement-job-to-push-story-to-Supabase-c12d2f9a3bdc4a24bc432d701c7c53f0

[//]: # "Add the relevant Notion link(s) here"

### Online sources

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"
N/A

### Related PRs
Related PRs will be Marco's HTML Parsing function that will be able to extract specific information from the content attribute such as the story content itself, process, and excerpt 

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review
1. Run getAllStories() to test output of fetch api on wordpress story endpoint
2. In order to test the functionality of the filterStories() function, we can console.log(indexList) to see how many objects at that given index will be removed. We can compare this to the number of stories that are being outputed by the getAllStories() function. 
3. Run node supabase.js to run the insertStoryData(); 
function which will call the getAllStories() and filterStories() function that both live in the index.js file. We can see the supabase stories schema getting updated with 5 out of 10 new inputs (5 were filtered out due to not being a text-based story object)
    
[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"
In terms of next steps, I need to clean up the insertStoryData() function to pull correct outputs to be placed in the schema. As of right now, the featured_media attribute only returns the specific story ID. I also need to find a way to clean up and parse the links that are nested in the object output. I also need to parse the HTML content output to correctly grab data for the excerpt and process values. However, the data flow and the filtering system is correct. 
## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @akshaynthakur

[//]: # "This tags in Akshay as a default. Feel free to change, or add on anyone who you should be in on the conversation."
